sklearn/semi_supervised/tests/test_self_training.py

```diff
@@ -0,0 +1,165 @@
+    # Check classification for various parameter settings.
+    # Also assert that predictions for strings and numerical labels are equal.
+    # Also test for multioutput classification
+    grid = ParameterGrid({"max_iter": [1, 50, 100],
+                          "threshold": [0.0, 0.5, 0.9]})
+
+    for base_classifier in [DummyClassifier(random_state=0),
+                            DecisionTreeClassifier(random_state=0),
+                            KNeighborsClassifier(),
+                            SVC(gamma="scale", probability=True,
+                                random_state=0)]:
+        for params in grid:
+            st = SelfTrainingClassifier(base_classifier, **params)
+            st_string = SelfTrainingClassifier(base_classifier, **params)
+            st.fit(X_train, y_train_missing_labels)
+            pred = st.predict(X_test)
+            proba = st.predict_proba(X_test)
+
+            st_string.fit(X_train, y_train_missing_strings)
+            pred_string = st_string.predict(X_test)
+            proba_string = st_string.predict_proba(X_test)
+
+            assert_array_equal(np.vectorize(mapping.get)(pred), pred_string)
+            assert_array_equal(proba, proba_string)
+
+
+def test_missing_predict_proba():
+    # Check that an error is thrown if predict_proba is not implemented
+    base_classifier = SVC(gamma="scale")
+    self_training = SelfTrainingClassifier(base_classifier)
+    message = "base_classifier (SVC) should implement predict_proba!"
+    assert_raise_message(ValueError, message, self_training.fit, X_train,
+                         y_train)
+
+
+def test_invalid_params():
+    # Test negative iterations
+    grid = ParameterGrid({"max_iter": [-1, -100, -10]})
+    base_classifier = SVC(gamma="scale", probability=True)
+    for params in grid:
+        st = SelfTrainingClassifier(base_classifier, **params)
+        message = "max_iter must be >= 0"
+        assert_raise_message(ValueError, message, st.fit, X_train, y_train)
+
+    grid = ParameterGrid({"threshold": [1.0, -2, 10]})
+    base_classifier = SVC(gamma="scale", probability=True)
+    for params in grid:
+        st = SelfTrainingClassifier(base_classifier, **params)
+        message = "threshold must be in [0,1)"
+        assert_raise_message(ValueError, message, st.fit, X_train, y_train)
+
+
+def test_single_iteration():
+    # Check classification for single iteration.
+    # Fitting a SelfTrainingClassifier with one iteration and 100 unlabeled
+    # datapoints should give the same results as fitting a normal classifier
+    # with only 50 labeled datapoints.
+
+    clf1 = SelfTrainingClassifier(KNeighborsClassifier(),
+                                  max_iter=0).fit(X_train,
+                                                  y_train_missing_labels)
+
+    clf2 = KNeighborsClassifier().fit(X_train[:limit], y_train[:limit])
+
+    assert_array_equal(clf1.predict(X_test), clf2.predict(X_test))
+
+
+def test_notfitted():
```

Comment:
This is legit but maybe there's a generic test for all estimators for this? Would need additional input.

Reply:
I'm not sure either.

Reply:
Yes, there is a test for this, but probably not for `META_ESTIMATORS`... With upcoming estimator tags, this should be sorted.

Reply:
I just run `pytest sklearn/tests/test_common.py -v -k SelfTrainingClassifier` yielding
```
sklearn/tests/test_common.py::test_estimators[SelfTrainingClassifier(base_estimator=LogisticRegression(C=1))-check_estimators_unfitted(strict_mode=True)] PASSED
```

So, please remove this test as it is already tested in `test_common.py`.